packages/client/src/views/SysAdmin/InformantSMSNotification/informantSMSNotification.tsx

```diff
@@ -0,0 +1,324 @@
+import { GenericErrorToast } from '@client/components/GenericErrorToast'
+import {
+  Event,
+  GetInformantSmsNotificationsQuery,
+  SmsNotification,
+  SmsNotificationInput,
+  ToggleInformantSmsNotificationMutation,
+  ToggleInformantSmsNotificationMutationVariables
+} from '@client/utils/gateway'
+import { find, lowerFirst } from 'lodash'
+import { Frame } from '@opencrvs/components/lib/Frame'
+import { Header } from '@client/components/Header/Header'
+import { Navigation } from '@client/components/interface/Navigation'
+import { buttonMessages, constantsMessages } from '@client/i18n/messages'
+import { Button } from '@opencrvs/components/lib/Button'
+import { useMutation, useQuery } from '@apollo/client'
+import { TOGGLE_INFORMANT_SMS_NOTIFICATION_MUTATION } from './mutations'
+import { NOTIFICATION_STATUS } from '@client/views/SysAdmin/Config/Application/utils'
+import { Toast } from '@opencrvs/components/lib/Toast'
+
+const ToggleWrapper = styled.div`
+  margin-left: 24px;
+`
+const Action = styled.div`
+  margin-top: 32px;
+`
+
+type SmsNotificationProps = {
+  items: SmsNotification[]
+}
+
+enum INotificationName {
+  birthInProgressSMS = 'birthInProgressSMS',
+  birthDeclarationSMS = 'birthDeclarationSMS',
+  birthRegistrationSMS = 'birthRegistrationSMS',
+  birthRejectionSMS = 'birthRejectionSMS',
+  deathInProgressSMS = 'deathInProgressSMS',
+  deathDeclarationSMS = 'deathDeclarationSMS',
+  deathRegistrationSMS = 'deathRegistrationSMS',
+  deathRejectionSMS = 'deathRejectionSMS'
+}
+
+type IState = { [key in INotificationName]: boolean }
+
+const InformantNotification = () => {
+  const intl = useIntl()
+  const isOnline = useOnlineStatus()
+  const { loading, error, data, refetch } =
+    useQuery<GetInformantSmsNotificationsQuery>(
+      GET_INFORMANT_SMS_NOTIFICATIONS,
+      {
+        fetchPolicy: 'no-cache'
+      }
+    )
+
+  const informantNotifitionsData = React.useMemo(() => {
+    return data?.informantSMSNotifications ?? []
+  }, [data])
+
+  const [informantSMSNotificationState, setInformantSMSNotificationState] =
+    React.useState<IState>({
+      birthInProgressSMS: true,
+      birthDeclarationSMS: true,
+      birthRegistrationSMS: true,
+      birthRejectionSMS: true,
+      deathInProgressSMS: true,
+      deathDeclarationSMS: true,
+      deathRegistrationSMS: true,
+      deathRejectionSMS: true
+    })
+
+  React.useEffect(() => {
+    const updateState = () => {
+      if (data && data.informantSMSNotifications) {
+        setInformantSMSNotificationState((state) => {
+          const modifiedState: IState = { ...state }
+          informantNotifitionsData.forEach((notification) => {
+            modifiedState[notification.name as INotificationName] =
+              notification.enabled
+          })
+          return {
+            ...state,
+            ...modifiedState
+          }
+        })
+      }
+    }
+    updateState()
+  }, [data, setInformantSMSNotificationState, informantNotifitionsData])
```

Comment:
Instead of this useEffect, we should be able to use the onCompleted event on the useQuery hook